pkg/products/grafana/reconciler.go

```diff
@@ -373,7 +373,7 @@ func (r *Reconciler) reconcileComponents(ctx context.Context, client k8sclient.C
 		if !k8serr.IsNotFound(err) {
 			return integreatlyv1alpha1.PhaseFailed, err
 		}
-		return integreatlyv1alpha1.PhaseAwaitingComponents, fmt.Errorf("waiting for prometheus service for grafana datasource to become available, %w", err)
+		return integreatlyv1alpha1.PhaseAwaitingComponents, nil
```

Comment:
I think we should still pass out the error in this case. We need to handle this error where ever it's been returned to.

Reply:
it returns it here https://github.com/integr8ly/integreatly-operator/blob/master/pkg/products/grafana/reconciler.go#L144-L148
and `err` get passed here https://github.com/integr8ly/integreatly-operator/blob/master/pkg/resources/events/events.go#L28-L31 
and never seen after 
Would it be better to log it as INFO/WARNING instead?

Reply:
Found an example of this phase -> https://github.com/integr8ly/integreatly-operator/blob/master/pkg/products/cloudresources/reconciler.go#L350-L352 
I assumed this is how we want to handle it

Reply:
The function for this message can only pass out an error or nil. It's up to the `HandleError` to decide what to do with the error message once it gets it. I still feel we should pass this message back out to the receiver. Maybe we need a second opinion on this!

Reply:
It makes sense, but should then all other `PhaseAwaitingComponents` be changed to return an error? For the name of consistency